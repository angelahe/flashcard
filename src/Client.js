// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(L1word, L2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  return fetch(`api/card?EN_word=${L1word}&ES_word=${L2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
