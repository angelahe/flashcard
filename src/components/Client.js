// interract with db back end

function createDeck() {
  return fetch('/api/deck', {
    method: 'POST',
  })
    .then(response => response.json())
    .then(json => json.id);
}

function createCard(deckId, l1word, l2word) {
  // let's try to send an api call with hardcoded values like in postman
  // return fetch('api/card?EN_word=test&ES_word=prueba', {
  // let's try to send the values passed in
  // 127.0.0.1:5000/api/deck/68dfd811-9913-422f-8c50-037055314e13/card?EN_word=test&ES_word=prueba
  console.log(`args are deck:${deckId}, l1word: ${l1word}, l2word: ${l2word}!`);
  return fetch(`api/deck/${deckId}/card?EN_word=${l1word}&ES_word=${l2word}`,
  //  return fetch(`api/deck/card?EN_word=${l1word}&ES_word=${l2word}`,
    {
    // return fetch('/api/card', {
      method: 'POST',
    })
    .then(response => response.json())
    .then(json => json.id);
}
const Client = { createDeck, createCard };
export default Client;
