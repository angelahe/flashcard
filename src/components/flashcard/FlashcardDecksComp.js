import React from 'react';
import PropTypes from 'prop-types';
import Client from '../Client';
import '../../styles/flashcard.css';
import DeckComp from './DeckComp';

class FlashcardDecksComp extends React.Component {
  _isMounted = false;

  constructor(props) {
    super(props);
    this.state = {
      deckList: [],
    };
  }

  componentDidMount = () => {
    this.loadDecks();
    this._isMounted = true;
  }

  componentWillUnmount = () => {
    this._isMounted = false;
  }

  loadDecks = () => {
    Client.getDecks()
      .then((decks) => {
        if (this._isMounted) {
          this.setState({ deckList: decks });
        }
      })
      .catch(err => console.error('There was a problem with getting decks', err.message));
  }

  handleDeckSelect = (deck) => {
    const { onDeckSelect } = this.props;
    console.log('select deck, need to render game select screen');
    onDeckSelect(deck);
  }

  render() {
    const { deckList } = this.state;

    const deckListItems = deckList.map(deck => (
      <DeckComp
        key={deck.deck_id}
        deck={deck}
        onDeckSelect={this.handleDeckSelect}
      />
    ));

    return (
      <div className="FlashcardDecksComp">
        <div className="DeckContainer">
          {deckListItems}
        </div>
      </div>
    );
  }
}
FlashcardDecksComp.defaultProps = {
  onDeckSelect: () => { },
};
FlashcardDecksComp.propTypes = {
  onDeckSelect: PropTypes.func,
};
export default FlashcardDecksComp;
