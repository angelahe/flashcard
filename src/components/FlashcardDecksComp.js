import React from 'react';
import '../styles/flashcard.css';

class FlashcardDecksComp extends React.Component {
  constructor() {
    super();
    this.state = {
      show: true,
    };
  }

  render() {
    return (
      <div className="FlashcardDecksComp">
        <p>FlashcardDecksComp coming soon</p>
      </div>
    );
  }
}
export default FlashcardDecksComp;
