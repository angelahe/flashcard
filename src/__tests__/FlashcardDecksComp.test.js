import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import FlashcardDecksComp from '../components/flashcard/FlashcardDecksComp';
import Client from '../components/Client';
import deckList from '../testdata/testdecklist';

const decks = JSON.stringify([
  { deck_id: '76ff91c7-8c30-4226-93f2-2c295fdb9939', deck_name: 'Animals' },
  { deck_id: '1a614b6b-408a-4365-94be-2b87be6b94dc', deck_name: 'Birds' },
]);

// const deck = { deck_id: 'ABCD-1234', deck_name: 'Animals' };

it('renders without crashing', () => {
  fetch.once(decks);
  shallow(<FlashcardDecksComp />);
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  fetch.once(decks);
  ReactDOM.render(<FlashcardDecksComp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot of UI renders consistently', () => {
  fetch.once(decks);
  const tree = renderer.create(<FlashcardDecksComp />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('contains the container div FlashcardDecksComp on shallow render', () => {
  fetch.once(decks);
  const component = shallow(<FlashcardDecksComp />);
  expect(component.html()).toContain('FlashcardDecksComp');
});

it('goes to ManageDecks on Add Click', () => {
  fetch.once(decks);
  const handleAddClick = jest.fn(() => {});
  const component = shallow(<FlashcardDecksComp onAdd={handleAddClick} />);
  component.instance().handleAddClick();
  expect(handleAddClick.mock.calls.length).toBe(1);
});

it('shows decks loaded', (done) => {
  const handleAddClick = jest.fn(() => {});
  Client.getDecks = jest.fn(() => Promise.resolve(deckList.deckList));
  const component = shallow(<FlashcardDecksComp onAdd={handleAddClick} />);

  setTimeout(() => {
    expect(Client.getDecks.mock.calls.length).toBe(1);
    expect(component.instance().state.deckList).toBe(deckList.deckList);
    done();
  });
});
