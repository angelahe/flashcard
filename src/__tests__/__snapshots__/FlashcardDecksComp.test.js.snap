// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of UI renders consistently 1`] = `
<div
  className="FlashcardDecksComp"
>
  <div
    className="ActionBar ActionBarComp"
  >
    <img
      alt="logo"
      className="AppLogo NavImg NavBtn"
      src="logo_FFFFFF.png"
    />
    <button
      className="AddBtn NavBtn"
      onClick={[Function]}
      type="button"
    >
      <img
        alt="Add"
        className="NavImg"
        src="add_FFFFFF.png"
      />
    </button>
    <span
      className="HeaderText"
    >
      Flashcards
    </span>
  </div>
  <div
    className="DeckContainer"
  />
</div>
`;
